docs/development/new-cloud-provider.md

```diff
@@ -0,0 +1,56 @@
+# Adding Cloud Providers
+This document provides an overview of how to integrate a new cloud provider into Gardener. Each component that requires integration has a detailed description of how to integrate it and the steps required.
+
+## Cloud Components
+Gardener is composed of 2 or more Kubernetes clusters:
+
+* Shoot: These are one more "workload" clusters, the regular Kubernetes clusters you have seen. They provide places for your workloads to run.
+* Seed: This is the "management" cluster. It provides management service for one or more Shoot cluster.
+
+These two clusters can run in the same cloud provider, but they do not need to. For example, you could run your Seed in AWS, while having one shoot in Azure, two in Google, two in Alicloud, and three in Packet.
+
+The Seed cluster deploys and manages the Shoot clusters. Importantly, for this discussion, the `etcd` data store backing each Shoot runs as workloads inside the Seed. Thus, to use the above example, the clusters in Azure, Google, Alicloud and Packet will have their worker nodes and master nodes running in those clouds, but the `etcd` clusters backing them will run as separate [deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) in the Seed Kubernetes cluster on AWS.
+
+This distinction becomes important when preparing to integrate a new cloud provider.
+
+## Gardener Cloud Integration
+Gardener integrates with cloud providers at the following key lifecycle elements:
+
+* Create/destroy/get/list machines for the Shoot
+* Create/destroy/get/list machines for the Seed
+* Backup/restore etcd for the Seed via writing files to and reading them from object storage
+
```

Comment:
Please also list the infrastructure preparation, and *optionally* provider-specific DNS integration.

Reply:
If you take a look at the [`CloudBotanist` interface](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/types.go#L25-L60) you see some methods like `DeployInfrastructure` and `DestroyInfrastructure`.
These methods are getting called during creation of shoot - for AWS and other public clouds they create VPCs, subnetworks, routes, and security groups into which the VMs (provisioned by MCM) and LBs (provisioned by CCM/KCM) are deployed later.
Not sure what is needed for packet, but I expect sth. similar.

For DNS: Users may use their cloud-specific DNS provider when integrate into Gardener. Example: Alicloud just integrated, but they cannot use AWS Route53, Google Cloud DNS or others. Hence, they integrated Alicloud DNS as a separate step (#393). This is optional.